src/TMGToolbox/inputOutput/importMatrixFromCSVThirdNormalized.py

```diff
@@ -15,149 +15,273 @@
-        cmd = obj.getDelCmd()
-        model.getCommander().addCommand(cmd)
-        model.getCommander().addCommand(None)
-
-
-# Create new matrix
-# set the type of OD matrix if is transit or no
-matrix = None
-if vehicleEID == 'transit':
-    matrix = GKSystem.getSystem().newObject("GKPedestrianODMatrix", model)
-    print("Adding transit OD matrix")
-else:
-    matrix = GKSystem.getSystem().newObject("GKODMatrix", model)
-matrix.setExternalId(matrixId)
-matrix.setName(matrixId)
-matrix.setStoreId( 2 ) # use external ID when storing
-matrix.setStoreType( 0 ) # store in the aimsun file
-matrix.setCentroidConfiguration(centroidConfiguration)
-matrix.setValueToAllCells(0.0)
-matrix.setEnableStore(True)
-
-if vehicleEID != 'transit':
-    sectionType = model.getType("GKVehicle")
-    if catalog.findByName(vehicleEID, sectionType) is None:
-        raise Exception(f"The specified vehicle type '{vehicleEID}' does not exist")
-    else:
-        vehicleType = catalog.findByName(vehicleEID, sectionType)
-        matrix.setVehicle(vehicleType)
-elif vehicleEID == 'transit':
-    sectionType = model.getType("GKPedestrianType")
-    if catalog.findByName("Pedestrian", sectionType) is None:
-        raise Exception(f"The specified vehicle type Pedestrian does not exist")
+
+def run_xtmf(parameters, model, console):
+    """
+    A general function called in all python modules called by bridge. Responsible
+    for extracting data and running appropriate functions.
+    """
+    outputNetworkFile = parameters["OutputNetworkFile"]
+    _execute(outputNetworkFile, model, console, parameters)
+
+def _execute(outputNetworkFile, inputModel, console, parameters):
+    """ 
+    Main execute function to run the simulation 
+    """
+    model = inputModel
+    catalog = model.getCatalog()
+    geomodel = model.getGeoModel()
+    
+    fileLocation = str(parameters["ODCSV"])
+    thirdNormalized = bool(parameters["ThirdNormalized"])
+    header = bool(parameters["IncludesHeader"])
+    matrixId = str(parameters["MatrixID"])
+    centroidConfigurationId = str(parameters["CentroidConfiguration"])
+    vehicleEID = str(parameters["VehicleType"])
+    initialTime = str(parameters["InitialTime"])
+    durationTime = str(parameters["DurationTime"])
+
+    #find the centroid configuration
+    sectionType = model.getType("GKCentroidConfiguration")
+    centroidConfiguration = catalog.findObjectByExternalId(centroidConfigurationId, sectionType)
+    if centroidConfiguration is None:
+        raise Exception(f"The specified centroid configuration '{centroidConfigurationId}' does not exist")
+    #check if matrix with given id exists and delete it if it does
+    if catalog.findObjectByExternalId(matrixId) != None:
+        while catalog.findObjectByExternalId(matrixId) != None:
+            obj = catalog.findObjectByExternalId(matrixId)
+            cmd = obj.getDelCmd()
+            model.getCommander().addCommand(cmd)
+            model.getCommander().addCommand(None)
```

Comment:
We should break this logic for deleting a matrix out into a separate method.

Reply:
I wasn't sure what these lines of code are actually doing so I left it as is. I could break it at a later point once I have all the modules running from the bridge?